src/Native/Unix/System.Native/pal_process.c

```diff
@@ -187,6 +197,20 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
         goto done;
     }
 
+#if !HAVE_PIPE2
+    // We do not have pipe2(); take the lock to emulate it race free.
+    // If another process were to be launched between the pipe creation and the fcntl call to set CLOEXEC on it, that
+    // file descriptor will be inherited into the other child process, eventually causing a deadlock either in the loop
+    // below that waits for that pipe to be closed or in StreamReader.ReadToEnd() in the calling code.
+    if (pthread_mutex_lock(&ProcessCreateLock) != 0)
+    {
+        // This check is pretty much just checking for trashed memory.
+        success = false;
+        goto done;
+    }
+    haveProcessCreateLock = 1;
```

Comment:
This should be `haveProcessCreateLock = true;`

Reply:
Done.